module/VuFind/src/VuFind/OAuth2/Entity/UserEntity.php

```diff
@@ -66,7 +66,26 @@ public function __construct(
         protected AccessTokenServiceInterface $accessTokenService,
         protected ILSAuthenticator $ilsAuthenticator
     ) {
-        $this->setIdentifier($user->getId());
+        $userIdentifierField = $oauth2Config['Server']['userIdentifierField'] ?? 'id';
+        switch ($userIdentifierField) {
+            case 'id':
+                $userIdentifier = $user->getId();
+                break;
+            case 'username':
+                $userIdentifier = $user->getUsername();
+                break;
+            case 'cat_id':
+                $userIdentifier = $user->getCatId();
+                break;
+            default:
+                $userIdentifier = null;
+        }
+        if ($userIdentifier === null) {
+            throw new \VuFind\Exception\BadConfig(
+                "Required user identifier field $userIdentifierField empty for user {$user->id}"
```

Comment:
Should we throw a different exception in the default case above? If an unsupported field is specified, that's not really an empty value error, so this message might be misleading.

Reply:
I updated the message. Hopefully it is more helpful now.